e2e/wdio/core/tests/dynamic-page.e2e-spec.ts

```diff
@@ -0,0 +1,156 @@
+import {
+    click,
+    doesItExist,
+    getAttributeByName,
+    getElementArrayLength,
+    getElementClass,
+    isElementDisplayed,
+    pause,
+    refreshPage,
+    scrollIntoView,
+    waitForElDisplayed,
+    waitForPresent
+} from '../../driver/wdio';
+import { DynamicPagePo } from '../pages/dynamic-page.po';
+
+describe('dynamic side content test suite', function () {
+    const dynamicPagePage = new DynamicPagePo();
+    const {
+        button, pinButton, exitButton, saveButton, closeButton, cancelButton, rejectButton, acceptButton,
+        collapseButton, collapsibleHeader, columnLayoutExample, tabsExample, defaultExample,
+        responsiveExample, tab, dynamicPageContent, tabsContent, dynamicPage, flexileColumn
+    } = dynamicPagePage;
+
+    beforeAll(() => {
+        dynamicPagePage.open();
+    }, 1);
+
+    beforeEach(() => {
+        refreshPage();
+        waitForPresent(dynamicPagePage.root);
+        waitForElDisplayed(dynamicPagePage.title);
+    }, 1);
+
+    it('should check open-close', () => {
+        checkPageOpenClose(defaultExample);
+        checkPageOpenClose(tabsExample);
+        checkPageOpenClose(columnLayoutExample);
+        checkPageOpenClose(responsiveExample);
+    });
+
+    it('should check collapsing area', () => {
+        checkCollapsingArea(tabsExample);
+        checkCollapsingArea(columnLayoutExample);
+        checkCollapsingArea(responsiveExample);
+    });
+
+    it('should check selecting tabs', () => {
+        click(tabsExample + button);
+        scrollIntoView(tabsContent, 1);
+        expect(getElementClass(tab, 1)).toContain('is-selected', 'tab is not highlited as selected after scroll to content');
+        expect(getElementClass(tab, 0)).not.toContain('is-selected', 'tab is selected, but should not');
+        expect(getElementClass(tab, 2)).not.toContain('is-selected', 'tab is selected, but should not');
+    });
+
+    it('should check changing size of buttons size', () => {
+        click(responsiveExample + button);
+        click(dynamicPage + button, 7);
+        const buttonsLength = getElementArrayLength(dynamicPage + button);
+        for (let i = 0; i < buttonsLength; i++) {
+            if (i !== 5 && i !== 6) {
+                expect(getElementClass(dynamicPage + button, i)).not.toContain('compact', 'size of the button still compact');
+            }
+        }
+        click(dynamicPage + button, 7);
+        for (let i = 0; i < buttonsLength; i++) {
+            if (i !== 5 && i !== 6) {
+                expect(getElementClass(dynamicPage + button, i)).toContain('compact', 'size of the button did not change to cozy');
+            }
+        }
+    });
+
+    it('should check 📍 button working', () => {
```

Comment:
need to also check that the pinned header is still visible after scrolling down (check the pin actually working)